Replace Add-Member with PSCustomObjects

DanMakes opened the pull request:
Using Measure-Command, it appears that using pscustomobject is consistently faster than New-Object / Add-Member. I've updated a couple of these instances in your code and just wanted you to see.

DanMakes committed:
First Pass - Updating PSCustomObjects

First Pass - Updating PSCustomObjects

AdhocAdam commented:
I had to try this out myself - ![compare](https://user-images.githubusercontent.com/6636040/146032050-88a0e0a5-27c5-444b-b806-14c880e31677.PNG)

And all I can say is thank you for raising this!
![omg](https://user-images.githubusercontent.com/6636040/146028668-4c23ccb2-3ffd-4097-a2d3-036d9a486e80.gif)

If you need any help or have any questions, let me know. I look forward to seeing the updates and testing this out in its entirety.

DanMakes committed:
Final - PSCustomObjects Update

Here's the final ps1 file for the improvement of the New-Object > Add-Member functionality. Enjoy!

DanMakes committed:
Delete psCustomObject.ps1

DanMakes committed:
Final - smletsexchangeconnector

This is the final pass.

AdhocAdam commented:
Looks great, thanks again!
